operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -0,0 +1,342 @@
+            queryParameters.add(COUNT_QUERY_PARAMETER, DEFAULT_RESOURCE_COUNT + "");
+        } else {
+            queryParameters.add(COUNT_QUERY_PARAMETER, countParameter.getValue().as(FHIR_INTEGER).getValue() + "");
+        }
+        // We use gt/lt here in an effort to be more liberal in terms of what we return
+        // https://ibm-watsonhealth.slack.com/archives/C14JTTR6C/p1614181836050500?thread_ts=1614180853.047300&cid=C14JTTR6C
+        Parameter startParameter = getParameter(parameters, START_QUERY_PARAMETER);
+        if (startParameter != null) {
+            queryParameters.add(DATE_QUERY_PARAMETER, STARTING_FROM + startParameter.getValue().as(FHIR_DATE).getValue());
+        }
+        Parameter endParameter = getParameter(parameters, END_QUERY_PARAMETER);
+        if (endParameter != null) {
+            queryParameters.add(DATE_QUERY_PARAMETER, UP_UNTIL + endParameter.getValue().as(FHIR_DATE).getValue());
+        }
+        Parameter sinceParameter = getParameter(parameters, SINCE_QUERY_PARAMETER);
+        if (sinceParameter != null) {
+            queryParameters.add(LAST_UPDATED_QUERY_PARAMETER, STARTING_FROM + sinceParameter.getValue().as(FHIR_INSTANT).getValue());
+        }
+        return queryParameters;
+    }
+
+    /**
+     * @return the list of patient subresources that will be included in the $everything operation, as provided by the user
+     * @throws FHIRSearchException 
+     */
+    protected List<String> getOverridenIncludedResourceTypes(Parameters parameters) throws FHIRSearchException {
+        List<String> typeOverrides = new ArrayList<>();
+        Parameter typesParameter = getParameter(parameters, TYPE_QUERY_PARAMETER);
+        if (typesParameter == null) {
+            return typeOverrides;
+        }
+        String typeOverridesParam = typesParameter.getValue().as(FHIR_STRING).getValue();
+        String[] typeOverridesList = typeOverridesParam.split(",");
+        for (String typeOverride : typeOverridesList) {
+            typeOverrides.add(typeOverride.trim());
+        }
+        return typeOverrides;
+    }
+
+    /**
+     * @return the list of patient subresources that will be included in the $everything operaetion
+     * @throws FHIRSearchException 
+     */
+    private List<String> getDefaultIncludedResourceTypes() throws FHIRSearchException {
+        List<String> resourceTypes = new ArrayList<>(CompartmentUtil.getCompartmentResourceTypes(PATIENT));
+        // TODO: Practitioner and Organization are not included in the getCompartmentReourceTypes() by default but it seems
+        // like a couple of good additional resources to include and they are even mentioned as examples of resources
+        // to include in the docs: https://www.hl7.org/fhir/operation-patient-everything.html 
+        // resourceTypes.add(Practitioner.class.getSimpleName());
+        // resourceTypes.add(Organization.class.getSimpleName());
+        return resourceTypes;
+    }
```

Comment:
You should only really need this one time.

Reply:
Yeah, I call this method from the constructor only.